### DISCUSSION TITLE
Implementing RFC 4806 (OCSP Extensions to IKEv2)

### DISCUSSION THREAD

**Discussion by jfhren-stormshield** [strongswan/strongswan#1845] [General]
Hello,
We are currently implementing RFC 4806 which describes how OCSP responses can be requested and sent through an IKEv2 tunnel. Our goal is to make a PR.
The implementation is working fine but I have some questions:
- The RFC defines a new certificate encoding type (14) used in CERTREQ and CERT payloads. In configuration, one can disable sending CERTREQ and/or CERT payloads. Should it also impact CERTREQ and CERT payloads related to OCSP ?
- To determine the behavior for a connection, we added a new token `<conn>.ocsp`. Allowed values are `never`, `request`, `reply` and `both`. If `never` is set, nothing is done. If `request` is set, a CERTREQ payload (14) is sent and CERT payload (14) are processed. If `reply` is set, CERTREQ payload (14) are processed and CERT payload (14) are sent. If `both` is set, it's a combination of `request` and `reply`. Does that look fine ? 
- When sending a CERTREQ payload (14) according to RFC 4806, it should contain hashes of known OCSP responder certificates from group (b). Since it's related to local configuration, should a new connection token be added to set such OCSP responder certificates ?

Thank you.

**Comment by strongX509** [comment]
I think the **`authorities`** section could be extended to support a new token:
```
<name>.ocsp_signers
```
which would allow to load one or even several OCSP signer certificates for a given CA.

An alternative without defining a new token would be to put the OCSP Signer Certificates in **`swanctl/x509ocsp`** and probably filtering the OCSP signer certificates if the CERTREQ is restricted to a given CA.

**Reply by jfhren-stormshield** [reply]
Sorry for the (very) late reply. I used the alternative and it works perfectly.
I did a PR with all the modification: https://github.com/strongswan/strongswan/pull/2016
Thank you for your help.
